### PR TITLE
ssh: add ErrDenied as a way to indicate auth termination 

### PR DESCRIPTION
**Pull request by maisem**

### CODE REVIEW

File: ssh/server.go [hunk 1]
```diff
@@ -397,12 +404,18 @@ func (l ServerAuthError) Error() string {
...
 }
 
-// ErrNoAuth is the error value returned if no
-// authentication method has been passed yet. This happens as a normal
-// part of the authentication loop, since the client first tries
-// 'none' authentication to discover available methods.
-// It is returned in ServerAuthError.Errors from NewServerConn.
-var ErrNoAuth = errors.New("ssh: no auth passed yet")
+var (
+	// ErrDenied is used to inform the client that access is deined and that no
```
Comment (bradfitz) [inline ssh/server.go]: say where it's used?

// ErrDenied can be returned from X, Y, Z to ...

File: ssh/server.go [hunk 2]
```diff
@@ -397,12 +404,18 @@ func (l ServerAuthError) Error() string {
...
+	// further attempt will be accepted on the connection.
+	ErrDenied = errors.New("ssh: access denied")
+
+	// ErrNoAuth is the error value returned if no
```
Comment (bradfitz) [inline ssh/server.go:416]: returned by what?

File: ssh/server.go [hunk 3]
```diff
@@ -469,6 +482,17 @@ userAuthLoop:
 					authErr = nil
 				}
 			}
+			if errors.Is(authErr, ErrDenied) {
```
Comment (bradfitz) [inline ssh/server.go]: why not move this down to line 677?

**Review by bradfitz — APPROVED** [review]

File: ssh/server.go (hunk 2 above)
Comment (maisem) [reply ssh/server.go:416]: This is an old comment and it says returned by what the end. Not changing

File: ssh/server.go (hunk 3 above)
Comment (maisem) [reply ssh/server.go]: I initially had it there, but moved it up because of something that I don't recall. Done.

**Pull request merged**